app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -116,6 +118,7 @@
     private int pokemonHP;
     private boolean pokeFlyRunning = false;
     private int trainerLevel;
+    private LruCache<String, String> OCRCache = new LruCache<>(200);
```

Comment:
> I had to add "name", "candy", "hp"

Ah I see. In a proper impl, it'd be cleaner and a bit more efficient to use a pair of the image hash and an enum for the key type, something like (pseudocode)
```
enum KeyType {NAME_KEY, CANDY_KEY, HP_KEY};
class Key { KeyType k; int imgHash; };
```
plus equals/hashcode methods, generated with an IDE. (I realize that's lots of code, which is why I prefer other JVM languages instead of Java).